compiler/rustc_hir/src/intravisit.rs

```diff
@@ -106,45 +106,41 @@ impl<'a> FnKind<'a> {
     }
 }
 
-/// An abstract representation of the HIR `rustc_middle::hir::map::Map`.
-pub trait Map<'hir> {
```

Comment:
If I understand correctly, this whole abstraction only exists to work around the fact that `Map`/`TyCtxt` isn't visible from the `rustc_hir` crate, and in fact every implementation is either (`rustc_middle::hir::Map` → `TyCtxt`) or `!`.

So it might be clearer to have the naming of this trait, its methods, and `nested_visit_map` reflect the fact that this is really just a `TyCtxt` in practice.

Reply:
That's exactly what I tried to do with the `HirCtxt`/`Cx` names. I'm happy to hear alternative suggestions.

Reply:
Ah, to be clear what I had in mind with my suggestion was a naming scheme like:

- `trait HirTyCtxt`
  - Every trait method would have exactly the same name as its real `TyCtxt` equivalent
- In the HIR visitor traits:
  - Associated `type TyCtxt`
  - `fn nested_visit_tcx()` (or even just `fn tcx()`)

The idea would be to signal to the reader “you can assume that this is really just a TyCtxt` under the hood”.

Reply:
Ok, I did a lot of that, though I went with `MaybeTyCtxt` for the associated type and `maybe_tcx` for the HIR visitor method.